cellengine/resources/fcs_file.py

```diff
@@ -1,15 +1,73 @@
 from __future__ import annotations
-from typing import List
-import fcsparser
+from cellengine.utils.dataclass_mixin import DataClassMixin
+from dataclasses import dataclass, field
+from dataclasses_json import config
+from typing import Any, Dict, List, Optional
+from fcsparser.api import FCSParser
 import pandas
+from pandas.core.frame import DataFrame
 
 import cellengine as ce
-from cellengine.payloads.fcs_file import _FcsFile
 from cellengine.resources.plot import Plot
 from cellengine.resources.compensation import Compensation
 
 
-class FcsFile(_FcsFile):
+@dataclass
+class FcsFile(DataClassMixin):
+    _id: str = field(metadata=config(field_name="_id"))
```

Comment:
A few fields need to be made read-only.

Reply:
More need to be made read-only -- all of the ones that were read-only in payloads/fcs_file.py.